swabbie-core/src/main/kotlin/com/netflix/spinnaker/swabbie/AbstractResourceTypeHandler.kt

```diff
@@ -305,9 +402,11 @@ abstract class AbstractResourceTypeHandler<T : Resource>(
     optedOutResourceStates: List<ResourceState>,
     action: Action
   ): Boolean {
+    if (resource.getViolations().any { summary -> summary.ruleName == "AlwaysCleanUpRule" }) return false
```

Comment:
Makes sure that if the `AlwaysCleanUpRule` is enabled (turned on in config file) that a resource isn't excluded from cleaning based on actual reasons (like it's in use). This is only for testing!

Reply:
These rule names should be constants.

Reply:
Oops. good catch